File: MATEMÁTICA/1429/main.dart
import 'dart:io';

int decimal(int acm) {
  List<int> factorial = [1, 1, 2, 6, 24, 120];
  int result = 0;
  for (int i = 1; acm > 0; i++) {
    int digit = acm % 10;
    result += digit * factorial[i];
    acm = acm ~/ 10;
  }
  return result;
}

void main() {
  int acm;
  while ((acm = int.parse(stdin.readLineSync()!)) > 0) {
    print(decimal(acm));
  }
}

File: MATEMÁTICA/1429/main.c
#include <stdio.h>

int decimal(int acm) {
    int factorial[6] = {1, 1, 2, 6, 24, 120};
    int result = 0;
    for (int i = 1; acm > 0; acm /= 10, i++) {
        int digit = acm % 10;
        result += digit * factorial[i];
    }
    return result;
}

int main() {
    int acm;
    scanf("%d", &acm);
    while (acm) {
        printf("%d\n", decimal(acm));
        scanf("%d", &acm);
    }
    return 0;
}
